common/chain/pubdata_helper.go

```diff
@@ -28,7 +28,7 @@ import (
 	"github.com/bnb-chain/zkbnb/types"
 )
 
-func ParseRegisterZnsPubData(pubData []byte) (tx *types.RegisterZnsTxInfo, err error) {
+func ParseRegisterZnsPubData(pubData []byte) (tx *legendTxTypes.RegisterZnsTxInfo, err error) {
```

Comment:
Can you help restructure the packages layout of `zkbnb-crypto` before move on this PR? `Legend` seems meanless in our solution.